client-app/shared/payment/components/payment-processing-skyflow.vue

```diff
@@ -100,9 +211,32 @@ async function pay() {
   loading.value = false;
 }
 
+async function pay() {
+  const res = (await composableContainer.collect({
+    additionalFields: getAdditionalRecords(),
+  })) as IInsertResponse;
+
+  if (!res?.records) {
+    emit("fail");
+  }
+
+  await processPayment(objectToKeyValue(res.records.find((el) => el.fields)?.fields as FieldsType));
+}
+
+watch(addNewCardSelected, async (value) => {
+  if (value) {
+    await initPayment();
+    createForm();
+  }
+});
```

Comment:
watch is calling twice if 
1. Select new card
2. Select saved card
3. Select new card